packages/flutter_tools/lib/runner.dart

```diff
@@ -65,6 +65,7 @@ Future<int> run(
     // in those locations as well to see if you need a similar update there.
 
     // Seed these context entries first since others depend on them
+    context.putIfAbsent(BotDetector, () => const BotDetector());
```

Comment:
This should go in the "Order-independent context entries" if possible.

Reply:
The armor plating makes it safe for this to go later, so, moved.